src/graphql/graphql.js

```diff
@@ -0,0 +1,108 @@
+/*******************************************************************************
+ *
+ *    Copyright 2018 Adobe. All rights reserved.
+ *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License. You may obtain a copy
+ *    of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software distributed under
+ *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ *    OF ANY KIND, either express or implied. See the License for the specific language
+ *    governing permissions and limitations under the License.
+ *
+ ******************************************************************************/
+
+'use strict';
+
+const request = require('request-promise-native');
+const { parse } = require('graphql');
+const { gqlToObject, makeGraphqlQuery } = require('@adobe/commerce-cif-graphql/utils');
+const MagentoClientBase = require('@adobe/commerce-cif-magento-common/MagentoClientBase');
+
+const ObjectTransformer = require('@adobe/commerce-cif-graphql/ObjectTransformer');
+const magentoTransformRules = require('./magentoTransformRules');
+
+const introspectionHandler = require('@adobe/commerce-cif-graphql/introspectionHandler');
+
+const ArgsTransformer = require('@adobe/commerce-cif-graphql/ArgsTransformer');
+const { transformerFunctions, checkFields } = require('./magentoArgsTransformer');
+const argsTransformer = new ArgsTransformer(transformerFunctions, checkFields, '__args');
+
+const ResponseMapper = require('@adobe/commerce-cif-graphql/ResponseMapper');
+const magentoMapper = require('./magentoMapper');
+
+function main(args) {
+    return introspectionHandler(args, magentoDataHandler);
```

Comment:
According to the API, `query` is a required parameter. We should use the `InputValidator` to return a proper error message in case the parameter is missing. I also don't see the check to be done inside the `introspectionHandler`.